internal/providers/providers.go

```diff
@@ -88,14 +84,10 @@ func WithRestClientCache(cache ratecache.RestClientCache) ProviderBuilderOption
 }
 
 // NewProviderBuilder creates a new provider builder.
-func NewProviderBuilder(
-	p *db.Provider,
-	ownerFilter sql.NullString,
-	credential provinfv1.Credential,
-	opts ...ProviderBuilderOption,
-) *ProviderBuilder {
+func NewProviderBuilder(p *db.Provider, ownerFilter sql.NullString, credential provinfv1.Credential, cfg *serverconfig.Config, opts ...ProviderBuilderOption) *ProviderBuilder {
 	pb := &ProviderBuilder{
 		p:           p,
+		cfg:         cfg,
```

Comment:
We'd have to pass the config everywhere when we create a new `ProviderBuilder`.

Reply:
Yeah, I don't think there's away around that.... unless we end up creating a `ProviderBuilderFactory` that has a set of persistent values like metrics and configuration. (screams in Java)